src/shared/components/create-case-filters/create-case-filters.component.ts

```diff
@@ -173,42 +157,41 @@ export class CreateCaseFiltersComponent implements OnInit {
   }
 
   private resetCaseType(): void {
-    this.resetErrors();
+    this.emitChange();
     this.filterCaseTypeControl.setValue('');
     this.selected.caseType = null;
     this.selectedJurisdictionCaseTypes = [];
     this.formGroup.controls['caseType'].disable();
   }
 
   private resetEvent(): void {
-    this.resetErrors();
+    this.emitChange();
     this.filterEventControl.setValue('');
     this.selected.event = null;
     this.selectedCaseTypeEvents = [];
     this.formGroup.controls['event'].disable();
   }
 
-  resetErrors(): void {
-    this.error = null;
-    this.ignoreWarning = false;
-    this.callbackErrorsSubject.next(null);
-    this.alertService.clear();
-  }
-
-  private hasCallbackErrors(): boolean {
-    return this.error
-      && this.error.callbackErrors
-      && this.error.callbackErrors.length;
-  }
-
-  private hasInvalidData(): boolean {
-    return this.error
-      && this.error.details
-      && this.error.details.field_errors
-      && this.error.details.field_errors.length;
+  emitChange(): void {
+    setTimeout(() => {
```

Comment:
Why setTimeout() here?

Reply:
To prevent "ExpressionChangedAfterItHasBeenCheckedError" errors. Couldn't come up with a more elegant way unfortunately.

Reply:
Ok. I see comment is there so ok.